### PR TITLE
all: use charm.URL not charm.Reference

### PR DESCRIPTION
**Pull request by rogpeppe**
This is the same as PR #45 except targeted at v2-unstable.
Note that the charmstore dependency is not yet part of the
master branch, as there's a cyclic dependency.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/110/
Test PASSed.

File: charmstore_test.go [hunk 1]
```diff
@@ -526,6 +521,7 @@ func (s *charmStoreRepoSuite) TestResolve(c *gc.C) {
 	}, {
 		id:  "~who/wordpress-2",
 		url: "cs:~who/precise/wordpress-2",
```
Comment (mhilton) [inline charmstore_test.go]: Remove this line, to make it clearer what the test does

**Comment by mhilton** [comment]
:+1: with a tiny nit

File: charmstore_test.go (hunk 1 above)
Comment (rogpeppe) [inline charmstore_test.go]: done

**Comment by frankban** [comment]
:+1: thank you!

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/111/
Test PASSed.

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**